cl/search/docket_number_cleaner.py

```diff
@@ -176,3 +241,206 @@ def clean_docket_number_raw(
                 return docket_number_raw, docket_id
 
     return docket_number_raw, None
+
+
+def create_llm_court_batches(
+    llm_batch: list[int],
+) -> dict[str, list[dict[int, str]]]:
+    """
+    Groups docket numbers by their associated court type for a given batch of docket_ids.
+
+    :param llm_batch: A list of docket_ids to be processed.
+    :return: A dictionary where each key is a court mapping (as determined by `court_map`),
+              and each value is a list of dictionaries of {docket_id: docket_number_raw}.
+    """
+    court_batches = {}
+    dockets = Docket.objects.filter(pk__in=llm_batch).values_list(
+        "id", "court_id", "docket_number_raw"
+    )
+    court_batches = defaultdict(list)
+    for docket_id, court_id, raw in dockets.iterator(chunk_size=2000):
+        if court_map.get(court_id) is not None:
+            court_batches[court_map.get(court_id)].append({docket_id: raw})
+    return court_batches
+
+
+def extract_with_llm(
+    batch: list[dict[int, str]], system_prompt: str, model_id: str
+) -> dict[int, str] | None:
+    """
+    Extracts and cleans a batch of docket numbers using a Large Language Model (LLM).
+
+    :param batch: The batch of records to process. Each item is a dictionary with the docket_id (int) as the key and the raw_docket_number (str) as the value.
+    :param system_prompt: The system prompt to provide to the LLM.
+    :param model_id: The identifier for the LLM model to use.
+    :return: A dictionary where keys are unique docket_ids (int) and values are cleaned docket numbers (str).
+    """
+    try:
+        llm_response = call_llm(
+            system_prompt=system_prompt,
+            user_prompt=f"{batch}",
+            model=model_id,
+            response_model=CleanDocketNumber,
+            temperature=0.1,
+            max_completion_tokens=1000,
+            api_key=env("DOCKET_NUMBER_CLEANING_API_KEY"),
+        )
+    except ValidationError as e:
+        logger.error(
+            "LLM - Response validation error for batch=%s",
+            f"{batch}",
+            extra={
+                "validation_errors": e.errors(),
+                "fingerprint": ["llm-docket-number-validation-error"],
+            },
+        )
+        return
+    except Exception as e:
```

Comment:
Since this is an API call, do we expect network errors, timeouts, or retriable HTTP errors (like 500) returned by the library?

I think it would be a good idea to catch these errors and return `None` instead of raising an exception. 

Otherwise, if one of the threads fails within `process_llm_batches`, all the work done by the successful threads might be lost.